src/component/plotly-graph/parse.js

```diff
@@ -155,11 +167,23 @@ function willFigureHang (result) {
 // can be (much) smaller than the true number of points plotted
 // when it does not match the length of the other coordinate arrays.
 function findMaxArrayLength (cont) {
-  const lengths = Object.keys(cont)
+  const arrays = Object.keys(cont)
     .filter(k => Array.isArray(cont[k]))
-    .map(k => cont[k].length)
+    .map(k => cont[k])
 
-  return Math.max(...lengths)
+  const lengths = arrays.map(arr => {
+    const innerArrays = arr.filter(Array.isArray)
```

Comment:
isn't this going to be super slow for big 1D arrays? Most if not all of our modules that support 1D or 2D differentiate just based on the first element.

Reply:
good call -> done in 
851b803